### ISSUE TITLE
Recurring error with Heat Exchanger when using cycling fan system

### ISSUE THREAD

**Issue by BobHenninger** [NREL/EnergyPlus#4298]
MJW's recommendation about user problem described below:
Heat exchanger needs to be smarter about cycling fan when deciding on the flow rate for both evaluating the effectiveness and for these warnings.

User is modelling an Energy Recovery Ventilator for a residential home.
 The model runs, however the warning: 
 "Average air volume flow rate is 130% of the nominal HX supply air
 volume flow rate" recurs several hundred thousand times.

MJW commented as follows:
These warnings always occur when the heat exchanger is running with a cycling fan system (something we should probably be able to suppress), so you can generally ignore them. Just to prove that to myself, I changed the fan operating mode schedule to be 1 for continuous fan all the time and reported on the system node volume flow rates and I did not see 1.3 m3/s everywhere.

 I fixed this by setting the unitary system design flow rates to 1.3 instead of 1.0 to match everything else in the model and get rid of this error:

 ** Warning ** Branch maximum flow rate differs from system flow rate.
 ** ~~~ ** Branch = BRANCH 1 has volume flow rate = 1.300000 m3/s.
 ** ~~~ ** System = AirLoopHVAC:Unitary:Furnace:HeatCool "FURNACE" has volume flow rate = 1.000000 m3/s.
 ** ~~~ ** A branch flow rate that is different from the system flow rate can cause discrepancies with outdoor air control.

 But I'm still not getting 1.3 everywhere. Most of the system is operating at 1.3, but there are times when the OA flow rates dips below that. And that's due to the HX going into defrost mode. If I turn that off, I get 1.3 all the time.

 So, I would recommend fixing the furnace flow rate value, but as long as you are using cycling fan, you will get these flow warnings from the hx.

Helpdesk Ticket 9344

User idf file
War Time Targets Test.idf

**Comment by rraustad** [comment]
Results:

```
Program Version,EnergyPlus, Version 8.3.4-3ec1754dc8, YMD=2015.09.20 15:23,IDD_Version 8.3.4
************* Testing Individual Branch Integrity
************* All Branches passed integrity testing
************* Testing Individual Supply Air Path Integrity
************* All Supply Air Paths passed integrity testing
************* Testing Individual Return Air Path Integrity
************* All Return Air Paths passed integrity testing
************* No node connection errors were found.
************* Beginning Simulation
************* Simulation Error Summary *************
************* There are 1 unused objects in input.
************* Use Output:Diagnostics,DisplayUnusedObjects; to see them.
************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
************* EnergyPlus Completed Successfully-- 0 Warning; 0 Severe Errors; Elapsed Time=00hr 01min 33.43sec
```

**Comment by rraustad** [comment]
Still not sure if the adjustments to flow and outlet conditions are correct. For a cycling fan, the air mass flow rate is the average and outlet conditions are full. For constant fan, the air mass flow rate is full and outlet conditions are averaged. So, for a cycling fan case, does using full outlet conditions in the OA system cause an issue?

```
if( FanOpMode == DataHVACGlobals::CycFanCycCoil ) {
    ExchCond( ExNum ).SupInMassFlow *= AirSidePLR;
    ExchCond( ExNum ).SupOutMassFlow *= AirSidePLR;
    ExchCond( ExNum ).SecInMassFlow *= AirSidePLR;
    ExchCond( ExNum ).SecOutMassFlow *= AirSidePLR;
    ExchCond( ExNum ).SupBypassMassFlow *= AirSidePLR;
    ExchCond( ExNum ).SecBypassMassFlow *= AirSidePLR;
} else if( FanOpMode == DataHVACGlobals::ContFanCycCoil ) {
    ExchCond( ExNum ).SupOutTemp = ExchCond( ExNum ).SupOutTemp * AirSidePLR + ExchCond( ExNum ).SupInTemp * ( 1.0 - AirSidePLR );
    ExchCond( ExNum ).SupOutHumRat = ExchCond( ExNum ).SupOutHumRat * AirSidePLR + ExchCond( ExNum ).SupInHumRat * ( 1.0 - AirSidePLR );
    ExchCond( ExNum ).SupOutEnth = ExchCond( ExNum ).SupOutEnth * AirSidePLR + ExchCond( ExNum ).SupOutEnth * ( 1.0 - AirSidePLR );
    ExchCond( ExNum ).SecOutTemp = ExchCond( ExNum ).SecOutTemp * AirSidePLR + ExchCond( ExNum ).SecInTemp * ( 1.0 - AirSidePLR );
    ExchCond( ExNum ).SecOutHumRat = ExchCond( ExNum ).SecOutHumRat * AirSidePLR + ExchCond( ExNum ).SecInHumRat * ( 1.0 - AirSidePLR );
    ExchCond( ExNum ).SecOutEnth = ExchCond( ExNum ).SecOutEnth * AirSidePLR + ExchCond( ExNum ).SecOutEnth * ( 1.0 - AirSidePLR );
}
```

![4298_results](https://cloud.githubusercontent.com/assets/5778156/9982504/e62e77c6-5fac-11e5-86a0-e1e59e9223c3.png)

**Comment by rraustad** [comment]
This is why there is no HX activity in summer:

```
HeatExchanger:AirtoAir:SensibleandLatent,
  HRV Heat Exchanger,      !- Name
  HRV_Schedule,            !- Availability Schedule Name
  1.3,                     !- Nominal Supply Air Flow Rate {m3/s}
  0.90,                    !- Sensible Effectiveness at 100% Heating Air Flow {dimensionless}
  0.90,                    !- Latent Effectiveness at 100% Heating Air Flow {dimensionless}
  0.90,                    !- Sensible Effectiveness at 75% Heating Air Flow {dimensionless}
  0.90,                    !- Latent Effectiveness at 75% Heating Air Flow {dimensionless}
  ,                        !- Sensible Effectiveness at 100% Cooling Air Flow {dimensionless}
  ,                        !- Latent Effectiveness at 100% Cooling Air Flow {dimensionless}
  ,                        !- Sensible Effectiveness at 75% Cooling Air Flow {dimensionless}
  ,                        !- Latent Effectiveness at 75% Cooling Air Flow {dimensionless}
```

**Comment by rraustad** [comment]
The desiccant HX does this differently but it looks like the same result.

A local variable for mass flow is increased to full flow:

```
if ( FanOpMode == CycFanCycCoil ) {
    SupInMassFlow /= HXPartLoadRatio; // supply = regen
    SecInMassFlow /= HXPartLoadRatio; // secondary = process
}
```

Then the outlet conditions are adjusted:

```
if ( FanOpMode == CycFanCycCoil || RegenInletIsOANode ) {
    //       Supply (regen) air stream mass flow rate is cycling and proportional to PLR, outlet conditions are full load conditions
    ExchCond( ExNum ).SupOutTemp = ExchCond( ExNum ).SupInTemp + FullLoadDeltaT;
    ExchCond( ExNum ).SupOutHumRat = min( 1.0, max( 1.e-5, ExchCond( ExNum ).SupInHumRat + FullLoadDeltaW ) );
} else {
    //       Supply (regen) air stream mass flow rate is constant and outlet conditions are averaged
    ExchCond( ExNum ).SupOutTemp = ExchCond( ExNum ).SupInTemp + ( FullLoadDeltaT * HXPartLoadRatio );
    ExchCond( ExNum ).SupOutHumRat = min( 1.0, max( 1.e-5, ExchCond( ExNum ).SupInHumRat + ( FullLoadDeltaW * HXPartLoadRatio ) ) );
}
```

**Comment by rraustad** [comment]
PR https://github.com/NREL/EnergyPlus/pull/5218

**Comment by rraustad** [comment]
I find it interesting that the OA flow rate is not proportional to the air loop flow. Even when it appears the controller is set up to provide an average flow rate proportional to the air loop branch flow rate. UPDATE: OK, this is due to defrost. Figure included below previous.

Defect file changed such that HVAC operates at same flow as other components. Also running only 1/1-1/10 and 7/1-7/10 in two run periods.

```
AirLoopHVAC:Unitary:Furnace:HeatCool,
  1.3,                   !- Supply Air Flow Rate During Cooling Operation {m3/s}
  1.3,                   !- Supply Air Flow Rate During Heating Operation {m3/s}

Controller:OutdoorAir,           
  OA Controller,
  HRV_inlet,               !- Relief Air Outlet Node
  Return_Air_NodeB,        !- Return Air Node
  Mixed_Air_Node,          !- Mixed Air Node
  OA_Intake_Node,          !- Actuator Node / OA inlet node
  1.3,                    !- Minimum Outdoor Air Flow Rate (%%was 0.03 from Hayes)
  1.3,                    !- Maximum Outdoor Air Flow Rate (%%was 0.03 from Hayes)
  NoEconomizer,            !- Economizer Control Type
  ModulateFlow,            !- Economizer Control Action Type
  19.,                     !- Economizer Maximum Limit Dry-Bulb Temperature {C}
  ,                        !- Economizer Maximum Limit Enthalpy {J/kg}
  ,                        !- Economizer Maximum Limit Dewpoint Temperature {C}
  ,                        !- Electronic Enthalpy Limit Curve Name
  4.,                      !- Economizer Minimum Limit Dry-Bulb Temperature {C}
  NoLockout,               !- Lockout Type
  FixedMinimum;            !- Minimum Limit Type
```

![oaversusmixedairmassflowrate](https://cloud.githubusercontent.com/assets/5778156/10026881/15b13e62-6132-11e5-93bf-18f8511a2093.png)

Flow rate when defrost is not active (still not sure why defrost affects OA flow rate if the OA controller is not controlling to a SA Temp SP and max/min flow are the same in the controller):

![4298_nodefost](https://cloud.githubusercontent.com/assets/5778156/10029486/2489a2b8-6140-11e5-9d84-5f30710136b4.png)

**Comment by rraustad** [comment]
A compromise was made in the OA system to operate the HX at a PLR = flow/max, for now, until this can be better understood.

```
if( FanOpMode == DataHVACGlobals::CycFanCycCoil ) {
    // HX's in the OA system can be troublesome given that the OA flow rate is not necessarily proportional to air loop PLR
    // adding that user input for branch flow rate, HX nominal flow rate, OA system min/max flow rate will not necessarily be perfectly input,
    // a compromise is used for OA sys HX's as the ratio of flow to max
//      AirloopPLR = AirLoopFlow( AirLoopNum ).FanPLR;
    AirloopPLR = OAController( OASysNum ).OAMassFlow / OAController( OASysNum ).MaxOAMassFlowRate;
} else {
    AirloopPLR = 1.0;
}
SimHeatRecovery( CompName, FirstHVACIteration, CompIndex, FanOpMode, AirloopPLR, _, _, _, AirLoopControlInfo( AirLoopNum ).HeatRecoveryBypass, AirLoopControlInfo( AirLoopNum ).HighHumCtrlActive );
```

**Comment by rraustad** [comment]
No HX warnings appear now even when Furnace flow is set back to 1.0 m3/s.
I think this is ready.

**Comment by rraustad** [comment]
PR https://github.com/NREL/EnergyPlus/pull/5218

**Comment by rraustad** [comment]
This answer looks reasonable. Effectiveness is a little higher than the user specified 0.9, which I don't have a good explanation for at this time.

![4298_effectiveness2](https://cloud.githubusercontent.com/assets/5778156/10030188/703f735a-6144-11e5-9877-114bbd673899.png)

**Comment by Myoldmopar** [comment]
Closed via #5218
